src/setup_scripts/upgrade_eth_token.rs

```diff
@@ -0,0 +1,170 @@
+use std::time::Duration;
+
+use starknet::accounts::{Account, ConnectedAccount};
+use starknet_providers::jsonrpc::HttpTransport;
+use starknet_providers::JsonRpcClient;
+use starknet_types_core::felt::Felt;
+use tokio::time::sleep;
+
+use crate::contract_clients::utils::{declare_contract, DeclarationInput, RpcAccount};
+use crate::helpers::account_actions::{get_contract_address_from_deploy_tx, AccountActions};
+use crate::utils::constants::{
+    EIC_ETH_TOKEN_CASM_PATH, EIC_ETH_TOKEN_SIERRA_PATH, NEW_ETH_TOKEN_CASM_PATH, NEW_ETH_TOKEN_SIERRA_PATH,
+};
+use crate::utils::wait_for_transaction;
+
+pub async fn upgrade_eth_token_to_cairo_1(
```

Comment:
can you write a comment on top of all upgrade functions explaining what's exactly happening. maybe we can copy from the notion